### ISSUE TITLE
flutter analyze --watch does not update when using build_runner

### ISSUE THREAD

**Issue by Parkar99** [flutter/flutter#67002]
I have noticed that since I have updated to 1.20.4, the analyzer is not updating with generated code. I have to restart the analysis server every time I generate new code with freezed, json_serializable, or any other generation libraries

**Comment by TahaTesser** [comment]
Hi @Parkar99 
Just tried `json_serializable` [example](https://pub.dev/packages/json_serializable/example) with `flutter analyze --watch` along `flutter pub run build_runner watch`
Generated are files updated along with analyzer too


### Test Video
[Screen Recording 2020-10-01 at 4.51.23 PM.zip](https://github.com/flutter/flutter/files/5311573/Screen.Recording.2020-10-01.at.4.51.23.PM.zip)




<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.20.4 at /Users/tahatesser/Code/flutter_stable
    • Framework revision fba99f6cf9 (2 weeks ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (1 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios • iOS 14.0.1

• No issues found!

```

</details>

**Comment by Parkar99** [comment]
Hello @TahaTesser 

I saw the video and yes it does seem to work. I did the same on my machine and it doesn't work.
I am running Ubuntu 20.04. I believe this is a Linux specific issue where `flutter analyze --watch` is not able to detect changes made by build_runner. The specific package I am using is freezed.

Here is my `flutter doctor` output.
```
[✓] Flutter (Channel stable, 1.20.4, on Linux, locale en_IN)
    • Flutter version 1.20.4 at /home/nabeel/bin/flutter
    • Framework revision fba99f6cf9 (2 weeks ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/nabeel/Android/Sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_265-8u265-b01-0ubuntu2~20.04-b01)
    • All Android licenses accepted.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 1 category.

```

FYI: I use NeoVIM and I used to use IntelliJ therefore the Android Studio issue. But that shouldn't affect the analyze

**Comment by TahaTesser** [comment]
Hi @Parkar99 
Just tried on the latest `Flutter (Channel stable, 1.22.0, on Linux, locale en_US.UTF-8)` on PopOS 20.04 based on Ubuntu 20.04 using [json_serializable example](https://pub.dev/packages/json_serializable/example)
```
No issues found! (3 fixed) • analyzed 2 files (ran in 0.00s)
```


<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.0, on Linux, locale en_US.UTF-8)
    • Flutter version 1.22.0 at /home/taha/Code/flutter_stable
    • Framework revision d408d302e2 (3 days ago), 2020-09-29 11:49:17 -0700
    • Engine revision 5babba6c4d
    • Dart version 2.10.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/taha/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /home/taha/Code/sdk
    • Java binary at: /home/taha/Code/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at /home/taha/Code/android-studio
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.15.0

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.

```

</details> 


I cannot reproduce this issue on the latest `stable` channel.
Can you please upgrade to the latest  `stable` and try again
`flutter channel stable`
`flutter upgrade --force`
`flutter doctor -v`

**Comment by Parkar99** [comment]
I am on 1.22. I upgraded yesterday after I made that post. I am still getting the same result. Since its not reproducable, I will try to reinstall the SDK. Thanks for your help
